addons/base_import/models/base_import.py

```diff
@@ -982,6 +1293,124 @@ def do(self, fields, columns, options, dryrun=False):
 
         return import_result
 
+    def _handle_multi_mapping(self, import_fields, input_file_data):
+        """ This method handles multiple mapping on the same field.
+        It will return the list of the mapped fields and the concatenated data for each field:
+            -> if two column are mapped on the same text or char field, they will end up
+            in only one column, concatenated via space (char) or new line (text).
+            -> The same logic is used for many2many fields. Multiple values can be imported if they are separated by ",".
+
+        Input/output Example:
+            input_file_data: [
+                ["Value part 1", "1", "res.partner_id1", "Value part 2"],
+                ["I am", "1", "res.partner_id1", "Batman"],
+            ]
+            import_fields: [desc, some_number, partner, desc]
+
+            output merged_data: [
+                ["Value part 1 Value part 2", "1", "res.partner_id1"],
+                ["I am Batman", "1", "res.partner_id1"],
+            ]
+            fields: [desc, some_number, partner]
+        """
+        # Get fields and their occurrences indexes
+        # Among the fields that have been mapped, we get their corresponding mapped column indexes
+        # as multiple fields could have been mapped to multiple columns.
+        mapped_field_indexes = collections.defaultdict(list)
```

Comment:
Is it necessary to use a default dict ? Always thought it was adding complexity without much added value compared to setdefault (when it is possible to use setdefault, obviously). Isn't something like the following working ?

```
mapped_field_indexes = {}
for idx, field in enumerate(field for field in import_fields if field):
    mapped_field_indexes.setdefault(field, list()).append(idx)
```